ghost/i18n/test/i18n.lint.js

```diff
@@ -0,0 +1,514 @@
+        sortedOverrideKeys.sort();
+
+        const newOverrides = {};
+        for (const rule of sortedOverrideKeys) {
+            newOverrides[rule] = newIgnoreFile.overrides[rule];
+        }
+
+        newIgnoreFile.overrides = newOverrides;
+
+        writeFileSync(this._ignoreFile.path, JSON.stringify(newIgnoreFile, null, 4));
+    }
+}
+
+/**
+ * Formats and logs the lint results. If there are any errors, the exit code will be 1
+ * @param {LintSummary} summary
+ */
+async function exitWithSummary(summary) {
+    const formatter = await new ESLint().loadFormatter();
+    const formattedResults = await formatter.format(summary.results, {cwd: '', rulesMeta: {}});
+
+    if (formattedResults) {
+        console.log(formattedResults); // eslint-disable-line no-console
+    }
+
+    if (summary.ignoreAllFlagProvidedWithoutFixFlag) {
+        console.warn( // eslint-disable-line no-console
+            '--unsafe-ignore-all was provided without --fix; errors were not ignored'
+        );
+    }
+
+    if (summary.errorCount > 0) {
+        console.log( // eslint-disable-line no-console
+            `\nNote: Since JSON doesn't support comments, use test/i18n-ignore.json to waive messages.\n\n`
+        );
+        process.exit(1);
+    }
+
+    process.exit(0);
+}
+
+/**
+ * Extracts translations variables and any invalid interpolations from a string
+ * @param {string} string the translation to parse
+ */
+function parseTranslationString(string) {
+    const START = 0;
+    const IN_VARIABLE = 1;
+    const errors = [];
+
+    let state = START;
+    let currentVariable = '';
+    let variableStartColumn = 0;
+    const variables = new Map();
+
+    // Use 1-indexed iterators since column is used a lot more than index
+    for (let column = 1; column <= string.length; column++) {
+        const character = string[column - 1];
+        if (character === '{') {
+            if (state === START) {
+                state = IN_VARIABLE;
+                // The brace is at the current column; the variable starts at the next column
+                variableStartColumn = column + 1;
+            } else {
+                errors.push({message: 'Unexpected "{" in variable', column});
+            }
+        } else if (character === '}') {
+            if (state === IN_VARIABLE) {
+                state = START;
+                variables.set(currentVariable, variableStartColumn);
+                currentVariable = '';
+            } else {
+                errors.push({message: 'Unexpected "}" in string', column});
+            }
+        } else if (state === IN_VARIABLE) {
+            currentVariable += character;
+        }
+    }
+
+    if (state === IN_VARIABLE) {
+        errors.push({message: 'Unclosed {', column: variableStartColumn});
+    }
+
+    return {variables, errors};
+}
+
+// TODO: should we glob all translations, or should we review the SUPPORTED_LOCALES?
```

Comment:
SUPPORTED_LOCALES should match with the filesystem folders, as there's a test for that.

An advantage of globs is that we don't have to update the test if someone adds or removes a namespace (which happened this week!)